apps/prairielearn/src/api/trpc/routers/course-files/updateQuestionFiles.ts

```diff
@@ -0,0 +1,64 @@
+import { z } from 'zod';
+
+import { IdSchema, QuestionSchema } from '../../../../lib/db-types.js';
+import { QuestionModifyEditor } from '../../../../lib/editors.js';
+import { selectCourseById } from '../../../../models/course.js';
+import { privateProcedure, selectUsers } from '../../trpc.js';
+
+export const updateQuestionFiles = privateProcedure
+  .input(
+    z.object({
+      // Context.
+      course_id: IdSchema,
+      user_id: IdSchema,
+      authn_user_id: IdSchema,
+      has_course_permission_edit: z.boolean(),
+      question: QuestionSchema,
```

Comment:
@jlevine18 rather than passing around an entire serialized question object, I think I'd prefer to pass around IDs (like `course_id` above). I know this might be a little wasteful when this is all running in the same process, but I generally feel better about just using IDs when crossing a network boundary. That means we don't have to worry about JSON (de)serialization of these objects. Thoughts?

Reply:
I was debating whether to apply the whole question (to fit with preexisting schemas) or just the qid (to save communication of unneeded information), so I'll just switch over.

Reply:
Switched in a063799.